### PR TITLE
Add audio media-type UI

### LINKED ISSUE #101: Add mute icon

### LINKED ISSUE #135: Microlink SDK: Supports for `audio` detection
**mini**

mini

**normal**

normal

**large**

large

<small>([See on Sketch](https://sketch.cloud/s/8AA73))</small>

**Related**

- https://medium.com/@erikras/%EF%B8%8F-audiocard-the-react-audio-player-designed-for-podcasts-and-twitter-cards-9311f9730e

**Inspiration**

- https://soundcloud.com/theaipodcast/gtc-weather
- https://anchor.fm/sin-humo/episodes/Episodio-9-Los-mandamientos-e22pro 
- https://open.spotify.com/album/7axhzfh8s79ObBSoyVfSIu
- https://www.youtube.com/watch?v=9P6rdqiybaw
- https://vimeo.com/186386161
- https://podcasts.apple.com/us/podcast/couples-therapy-with-candice-and-casey/id1380252026
- https://combine.fm/spotify/track/2w8i0VT0Gu4jfzBzpGPtK1
- https://transistor.fm/

### PR DESCRIPTION
**Pull request by breadadams**
Closes #135 and closes #101

**TODO**

- [x] Setup good `audio` demo urls.
- [x] Setup minimalist icons.
- [x] Setup monospace font for audio duration.
- [x] Show `Pause` button on hover.
- [x] Review all transition functions.
- [x] Only show `play` button at first impression for audio/video.
- [x] Implement `showControls` prop.

**Bugfix**

- [ ] ~Ensure video is not playing with `autoPlay={false}`~

**After Merge**

- [ ] ~Add `showControls` into docs.~

**Inspiration**

- https://iframely.com/docs/promo-cards
- https://share.transistor.fm/s/2c66c0a3

### CODE REVIEW

File: packages/react/stories/data.js [hunk 1]
```diff
@@ -1,29 +1,14 @@
...
-  'https://www.engadget.com/es/2017/10/23/meizu-m6-note-analisis-review-fotos/',
-  'https://www.nytimes.com/2017/09/19/learning/whats-going-on-in-this-graph-sept-19-2017.html',
-  'https://gizmodo.com/drone-video-of-border-wall-prototypes-accidentally-show-1819710328',
-  'https://www.bbc.com/news/technology-40762328',
-  'https://browserless.js.org',
-  'https://xn--vi8hl0c.ws/',
-  'https://media.giphy.com/media/LqTSLCsIIkCTvQ8X9g/giphy.gif'
-]
+import demoLinks from '@microlink/demo-links'
+import filter from 'lodash/filter'
```
Comment (breadadams) [inline packages/react/stories/data.js]: To just use 1 lodash util why not use `Array.filter()` @Kikobeats?

Comment (breadadams) [reply packages/react/stories/data.js]: Oh! 🤦‍♂

File: packages/react/src/components/Card/CardMedia/Wrap.js [hunk 2]
```diff
@@ -24,12 +24,11 @@ const mediaSizeStyles = {
 }
 
 export default styled('div')`
-  background: transparent no-repeat center center / cover;
+  background: #e1e8ed no-repeat center center / cover;
   display: block;
   overflow: hidden;
   height: auto;
   position: relative;
-  transition: flex-basis 0.25s ease-in-out;
```
Comment (breadadams) [inline packages/react/src/components/Card/CardMedia/Wrap.js]: How come you removed this @Kikobeats?

File: packages/react/src/components/Card/CardMedia/Video.js [hunk 3]
```diff
@@ -25,6 +27,26 @@ const Video = styled('video')`
...
+  .microlink_card:not(:hover) & {
+    opacity: ${({ visible }) => (visible ? 1 : 0)};
+  }
+
+`
+
+const VideoProgressBar = styled(ProgressBar)(
+  props => `
+  .microlink_card:hover & {
+    height: ${getProgressBarSize(props)}px;
```
Comment (breadadams) [inline packages/react/src/components/Card/CardMedia/Video.js]: I don't think we need this @Kikobeats, it doesn't change the height.

File: packages/react/src/components/Card/CardMedia/Video.js [hunk 4]
```diff
@@ -39,14 +61,15 @@ function CardVideo (props) {
     playsInline,
     ...restProps
   } = props
-  const [playing, setPlaying] = useState(autoPlay)
+  const [isPlaying, setPlaying] = useState(autoPlay)
```
Comment (breadadams) [inline packages/react/src/components/Card/CardMedia/Video.js]: I think it's good practice for the _setter_ to match the value, eg. `setIsPlaying`

File: packages/react/src/components/Card/CardMedia/controls/SeekButton.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
+import React, { useMemo } from 'react'
+import styled, { css } from 'styled-components'
+
+import MediaButton from './MediaButton'
+import { isLarge, media } from '../../../../utils'
+
+const Backward = props => (
```
Comment (breadadams) [inline packages/react/src/components/Card/CardMedia/controls/SeekButton.js]: @Kikobeats the rewind button says `30` but it goes back `10` seconds

File: packages/react/src/components/Card/CardMedia/Wrap.js (hunk 2 above)
Comment (Kikobeats) [reply packages/react/src/components/Card/CardMedia/Wrap.js]: I didn't note it's necessary, that transition is only visible on viewport resize, right?

File: packages/react/src/components/Card/CardMedia/Video.js (hunk 4 above)
Comment (Kikobeats) [reply packages/react/src/components/Card/CardMedia/Video.js]: true, I forgot that!

File: packages/react/src/components/Card/CardMedia/Wrap.js (hunk 2 above)
Comment (breadadams) [reply packages/react/src/components/Card/CardMedia/Wrap.js]: Yeah correct, I don't usually add those things either tbh when it's just a resize transition. Let's bin it 👍

File: packages/react/src/components/Card/CardMedia/Audio.js [hunk 6]
```diff
@@ -33,7 +30,7 @@ const Audio = ({
         autoPlay={autoPlay}
         cardSize={cardSize}
         loop={loop}
-        MediaComponent={AudioDOM}
+        MediaComponent='audio'
```
Comment (breadadams) [inline packages/react/src/components/Card/CardMedia/Audio.js]: LOL - why the heck didn't I do that 😆

File: packages/react/src/components/Card/CardMedia/Controls/SeekButton.js [hunk 7]
```diff
@@ -5,13 +5,19 @@ import MediaButton from './MediaButton'
 import { isLarge, media } from '../../../../utils'
 
 const Backward = props => (
-  <svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 29' {...props}>
+  <svg
+    xmlns='http://www.w3.org/2000/svg'
+    width='24'
+    height='29'
```
Comment (breadadams) [inline packages/react/src/components/Card/CardMedia/Controls/SeekButton.js]: We don't need the width/height attributes on the SVGs @Kikobeats

Comment (Kikobeats) [reply packages/react/src/components/Card/CardMedia/Controls/SeekButton.js]: removed!

**Pull request merged**